lib/src/react16_suggestors/pubspec_react_upgrader.dart

```diff
@@ -0,0 +1,92 @@
+// Copyright 2019 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import 'package:codemod/codemod.dart';
+import 'package:pub_semver/pub_semver.dart';
+import 'package:source_span/source_span.dart';
+
+/// Suggestor that attempts to update `pubspec.yaml` files to ensure a safe
+/// minimum bound on the `react` dependency.
+///
+/// If `react` is already listed, but the minimum bound is not high enough,
+/// the version constraint will be updated. If `react` is missing from
+/// the file, it will be added.
+class PubspecReactUpdater implements Suggestor {
+  /// Regex that matches the dependency constraint declaration for react.
+  static final RegExp reactDep = RegExp(
+    r'''^\s*react:\s*["']?([\d\s<>=^.]+)["']?\s*$''',
+    multiLine: true,
+  );
+
+  /// Regex that matches the `dependencies:` key in a pubspec.yaml.
+  static final RegExp dependenciesKey = RegExp(
+    r'^\s*dependencies:$',
+    multiLine: true,
+  );
+
+  final VersionRange targetConstraint;
+
+  PubspecReactUpdater(this.targetConstraint);
+
+  @override
+  Iterable<Patch> generatePatches(SourceFile sourceFile) sync* {
+    final contents = sourceFile.getText(0);
+    final reactMatch = reactDep.firstMatch(contents);
+    if (reactMatch != null) {
+      // react is already in pubspec.yaml
+      final line = reactMatch.group(0);
+      final constraintValue = reactMatch.group(1);
+      final constraint = VersionConstraint.parse(constraintValue);
+      if (constraint is VersionRange && constraint != targetConstraint) {
+        // Wrap the new constraint in quotes if required.
+        var newValue = targetConstraint.toString();
+        if (newValue.contains(' ') &&
+            !line.contains("'") &&
```

Comment:
Not sure I'm following here; doesn't `reactMatch`'s range, which we're replacing, include any existing quotes? So, it shouldn't matter whether they existed before since we're always removing them

Reply:
Yep, that is correct! The weirdness comes from `line.replaceFirst( constraintValue,newValue, ),` (full reference [here](https://github.com/Workiva/over_react_codemod/blob/d5024226acb6e169d1781afa0b9ef07ccfc94075/lib/src/react16_suggestors/pubspec_react_upgrader.dart#L101)). The match was matching the whole line, but that snippet was being used to target the constraint value itself (not including quotations).

That was weird though, so I updated both the react updater and over_react updater to work as you both expected 😄

Reply:
Heads up @greglittlefield-wf and @kealjones-wk, I ended up changing it back to the way it was. Removing it silently broke the codemod because there wasn't any whitespace before the text. That whitespace was easy enough to add, but the testing utils couldn't figure it out (even though an actual QA worked as expected). So after trying to get the tests to work correctly for a bit, it just seemed easier to add it back in.